ingest/src/main/java/com/microsoft/azure/kusto/ingest/ResourceManager.java

```diff
@@ -346,16 +358,55 @@ protected String retrieveServiceType() {
         return null;
     }
 
+    @Override
+    public void reportIngestionResult(ResourceWithSas<?> resource, boolean success) {
+        if (storageAccountSet == null) {
+            log.warn("StorageAccountSet is null");
+        }
+        storageAccountSet.addResultToAccount(resource.getAccountName(), success);
+    }
+
+    enum ResourceType {
+        SECURED_READY_FOR_AGGREGATION_QUEUE("SecuredReadyForAggregationQueue"),
+        FAILED_INGESTIONS_QUEUE("FailedIngestionsQueue"),
+        SUCCESSFUL_INGESTIONS_QUEUE("SuccessfulIngestionsQueue"),
+        TEMP_STORAGE("TempStorage"),
+        INGESTIONS_STATUS_TABLE("IngestionsStatusTable");
+
+        private final String resourceTypeName;
+
+        ResourceType(String resourceTypeName) {
+            this.resourceTypeName = resourceTypeName;
+        }
+
+        public static ResourceType findByResourceTypeName(String resourceTypeName) {
+            for (ResourceType resourceType : values()) {
+                if (resourceType.resourceTypeName.equalsIgnoreCase(resourceTypeName)) {
+                    return resourceType;
+                }
+            }
+            throw new IllegalArgumentException(resourceTypeName);
+        }
+
+        String getResourceTypeName() {
+            return resourceTypeName;
+        }
+    }
+
     private static class IngestionResource<T> {
+        final ResourceType resourceType;
```

Comment:
Is `T` not the `ResourceType`?

Reply:
It looks like this entire class is now useless with the new algorithm.

In fact there's seem to be a lot of dead code in ResourceManager, i'll remove it. Good catch.

Reply:
Upon second read - this class is actually important. ResourceType is an enum with the name of the resource, and exists here to map between the name (we get from kusto) to the resource itself. T is the actual resource stored in the array. We can't really change it without a full re-write